hack/run.sh

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env bash 
+
+set -e
+
+ROOT_DIR=$(dirname $0)/..
+
+action="$1"
+
+function usage() {
+	cmd="$1"
+
+	echo ""
+	echo "Usage                                                for more information read DEVELOPMENT.md"
+	echo "$cmd <command>"
+	echo ""
+	echo "command:"
+	echo "   install                                           Install Knative components for local dev"
+	echo "   e2e-debug <test-name> <test-dir>                  Quick debug"
+	echo "   update-cert-manager                               Update Cert manager"
+	echo "   generate-yamls <repo-root-dir> <output-file>      Generate all repo yamls "
+	echo "   update-reference-docs                             Update reference docs"
+	echo "   teardown                                          Teardown installed components for local dev"
+	echo "   update-checksums                                  Update checksums"
+	echo "   update-codegen                                    Update codegen"
+	echo "   update-deps                                       Update deps"
+	echo "   verify-codegen                                    Verify codegen"
+	echo ""
+}
+
+if [[ "${action}" == "install" ]]; then
+	source "${ROOT_DIR}"/hack/install.sh 
+elif [[ "${action}" == "update-cert-manager" ]]; then
+	source "${ROOT_DIR}"/hack/update-cert-manager.sh "$2" "$3"
+elif [[ "${action}" == "e2e-debug" ]]; then
+	source "${ROOT_DIR}"/hack/e2e-debug.sh "$2" "$3"
+elif [[ "${action}" == "generate-yamls" ]]; then
+	source "${ROOT_DIR}"/hack/generate-yamls.sh "$2" "$3"
+elif [[ "${action}" == "update-reference-docs" ]]; then
+	source "${ROOT_DIR}"/hack/update-reference-docs.sh
+elif [[ "${action}" == "teardown" ]]; then
+	source "${ROOT_DIR}"/hack/teardown.sh
+elif [[ "${action}" == "update-checksums" ]]; then
+	source "${ROOT_DIR}"/hack/update-checksums.sh
+elif [[ "${action}" == "update-codegen" ]]; then
+	source "${ROOT_DIR}"/hack/update-codegen.sh
+elif [[ "${action}" == "update-deps" ]]; then
+	source "${ROOT_DIR}"/hack/update-deps.sh 
+elif [[ "${action}" == "verify-codegen" ]]; then
+	source "${ROOT_DIR}"/hack/verify-codegen.sh
```

Comment:
Here, instead of sourcing the bash scripts could you directly execute them? E.g. instead of:
```bash
source "{ROOT_DIR}"/hack/script.sh
```
Simply:
```
./"{ROOT_DIR}"/has/script.sh
```

This is probably simpler and more in line with how people currently use the scripts, and it is similar to what we do in https://github.com/knative-extensions/eventing-kafka-broker/blob/7a3464cca004edecc57b8eb3fc177fe2969546c6/hack/run.sh#L90-L103
when there are scripts that we execute in their entirety rather than just calling a function defined within.